lib/CSharp2ClientGenerator.php

```diff
@@ -1139,6 +1142,83 @@ function writeMainClient(DOMNodeList $serviceNodes, DOMNodeList $configurationNo
 		$this->addFile("KalturaClient/Client.cs", $this->getTextBlock());
 	}
 
+	function writeRequestBuilderConfigurationClass($requestConfigurationNodes)
+	{
+		$this->startNewTextBlock();
+
+		$this->appendLine("using System;");
+		$this->appendLine("using Kaltura.Types;");
+		$this->appendLine("using Kaltura.Enums;");
+		$this->appendLine();
+
+		$this->appendLine("namespace Kaltura.Request");
+		$this->appendLine("{");
+		$this->appendLine("	public static class RequestBuilderExtensions");
+		$this->appendLine("	{");
+		foreach($requestConfigurationNodes as $configurationPropertyNode)
+		{
+			if ($configurationPropertyNode->nodeType != XML_ELEMENT_NODE)
+				continue;
+
+			$configurationProperty = $configurationPropertyNode->localName;
+
+			if($configurationPropertyNode->hasAttribute('volatile') && $configurationPropertyNode->getAttribute('volatile'))
```

Comment:
Since these properties will be set only to a single request, you should behave with volatile properties the same way as all other properties.